src/python/zquantum/core/wip/circuits/_wavefunction_operations.py

```diff
@@ -0,0 +1,46 @@
+from dataclasses import dataclass
+from numbers import Number
+
+from ._gates import _sub_symbols, Parameter
+from typing import Tuple, Sequence
+import numpy as np
+
+
+@dataclass(frozen=True)
+class MultiPhaseOperation:
+    """ TODO """
```

Comment:
Should this TODO be _todone_?

Reply:
Just some ideas: it'd be cool to some reference the origins of this gate / any external resource a future developer might use to validate that the implementation works as expected.

Reply:
While we are not aware of any literature covering this operation, I added a description to the docstring in 4aabd68
@alexjuda and @mstechly please see if it is correct and sufficient.

Reply:
Looks good to me!